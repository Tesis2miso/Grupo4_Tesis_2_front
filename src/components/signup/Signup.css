.signUpApp {
  text-align: center;
  background-color: rgb(26, 53, 235);
  background-image: url(/public/back.jpeg);
}

.signup {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  background-image: url(/public/back.jpeg);
  }

.title {
  color: white;
}

#shadowSignup {
  background: rgba(47, 54, 64, 0.5);
  border-radius: 12px;
  align-items: center;
  width: 500px;
  height: 800px;
  position: absolute;
}

.subtitle {
  color: lightgray;
}

input.formField {
  margin: 10px 0;
  width: 90%;
  height: 50px;
  font-size: 15px;
  border-radius: 10px;
  border: 1px solid lightgray;
  padding-left: 15px;
  padding-right: 15px;
}

.label_error {
  font-size:small;
  color: red;
}

.submitButton {
  background-color: #0053CE;
  color: white;
  font-size: 15px;
  width: 120px;
  height: 50px;
  border: 1px solid white;
  border-radius: 10px;
}

.conditionsLabel {
  font-size: 15px;
  margin-left: 10px;
}

.goToLogin {
  background-color: transparent;
  border: none;
  margin-bottom: 20px;
  font-size: 15px;
  color: lightgray;
}