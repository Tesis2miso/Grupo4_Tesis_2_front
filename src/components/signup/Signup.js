import "./Signup.css";
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useState, Fragment } from 'react';
import axios from "axios";

function Signup(props) {
    const navigate = useNavigate()
    const { loggedIn } = props
    useEffect(() => {
        if (loggedIn) {
            navigate('/')
        }
    })
    const [signupForm, setSignupForm] = useState({
        name: "",
        lastName: "",
        email: "",
        emailConfirmation: "",
        username: "",
        password: "",
        passwordConfirmation: "",
        checked: false
    })
    const [error, setError] = useState();

    const handleChange = (event) => {
        const { value, name } = event.target
        setSignupForm(prevNote => ({
            ...prevNote, [name]: value
        }))
    }

    const handleCheck = (event) => {
        const { checked } = event.target
        setSignupForm(prevNote => ({
            ...prevNote, checked: checked
        }))
    }

    document.addEventListener('DOMContentLoaded', () => {
        const miInput = document.getElementById('password');

        miInput.addEventListener('keyup', function (event) {
            if (event.getModifierState('CapsLock')) {
                alert("Bloq Mayús esta activado");
            }
        });
    });

    const signMeUp = (event) => {
        event.preventDefault()
        const {
            name, lastName,
            email, emailConfirmation,
            username, checked,
            password, passwordConfirmation
        } = signupForm

        if(email.trim() !== emailConfirmation.trim()) {
            setError('Emails does not match');
            return;
        }

        if (password.trim() !== passwordConfirmation.trim()) {
            setError('Passwords does not match');
            return;
        }

        if (!checked) {
            setError('You need to accepts terms and conditions');
            return;
        }

        axios({
            method: "POST",
            url: `${process.env.REACT_APP_BASE_PATH}/specialist`,
            data: {
                name: name,
                last_name: lastName,
                email: email,
                username: username,
                password: password,
            }
        })
            .then((response) => {
                props.setToken(response.data.token);
                navigate('/')
            }).catch((error) => {
                setError(error.response.data.mssg);
            })
    }

    const goToLogin = (event) => {
        event.preventDefault()
        navigate('/login')
    }

    return (
        <Fragment>
            <form className="signup">
                <h1 className="title">DermoApp</h1>
                <h2 className="subtitle">Registro de usuarios</h2>
                <br></br>
                <input
                    className="formField"
                    id="name"
                    onChange={handleChange}
                    type="text"
                    name="name"
                    placeholder="Name"
                    text={signupForm.name}
                    value={signupForm.name} />

                <input
                    className="formField"
                    id="lastName"
                    onChange={handleChange}
                    type="text"
                    name="lastName"
                    placeholder="Last name"
                    text={signupForm.lastName}
                    value={signupForm.lastName} />

                <input
                    className="formField"
                    id="email"
                    onChange={handleChange}
                    type="email"
                    name="email"
                    placeholder="Email"
                    text={signupForm.email}
                    value={signupForm.email} />

                <input
                    className="formField"
                    id="emailConfirmation"
                    onChange={handleChange}
                    type="email"
                    name="emailConfirmation"
                    placeholder="Email confirmation"
                    text={signupForm.emailConfirmation}
                    value={signupForm.emailConfirmation} />

                <input
                    className="formField"
                    id="username"
                    onChange={handleChange}
                    type="text"
                    name="username"
                    placeholder="Username"
                    text={signupForm.username}
                    value={signupForm.username} />

                <input
                    className="formField"
                    id="password"
                    onChange={handleChange}
                    type="password"
                    name="password"
                    placeholder="Password"
                    text={signupForm.password}
                    value={signupForm.password} />

                <input
                    className="formField"
                    id="passwordConfirmation"
                    onChange={handleChange}
                    type="password"
                    name="passwordConfirmation"
                    placeholder="Password confirmation"
                    text={signupForm.passwordConfirmation}
                    value={signupForm.passwordConfirmation} />

                <div className="form-check">
                    <input
                        onChange={handleCheck}
                        id="termsAndConditions"
                        type="checkbox"
                        className="checkTermsAndConditions"
                        checked={signupForm.checked}
                    />
                    <label className="conditionsLabel" htmlFor="termsAndConditions">
                        Do you accept terms and conditions?
                    </label>
                </div>

                <label className='label_error' id="error">{error}</label>
                <button id="goToLogin" className="goToLogin" onClick={goToLogin}>¿Ya tienes cuenta? Iniciar sesión</button>
                <button id="submitbtn1" className="submitButton" onClick={signMeUp}>Registrarse</button>
            </form>
        </Fragment>
    );
}

export default Signup;
