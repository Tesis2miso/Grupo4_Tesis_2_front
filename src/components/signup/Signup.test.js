import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Signup from './Signup';
import axios from "axios";
import { render } from '@testing-library/react';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
jest.mock('axios', () => jest.fn());

describe('Tests_Signup', () => {

  var wrapper = null;
  var non_logged_in_wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<Signup loggedIn={true} setToken={() => { }} />);
    non_logged_in_wrapper = shallow(<Signup loggedIn={false} setToken={() => { }} />);
  });

  test('show signup', () => {
    render(<Signup loggedIn={true} setToken={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('show signup inputs if logged in', () => {
    expect(wrapper.find('#name').text()).toEqual("");
    expect(wrapper.find('#lastName').text()).toEqual("");
    expect(wrapper.find('#email').text()).toEqual("");
    expect(wrapper.find('#emailConfirmation').text()).toEqual("");
    expect(wrapper.find('#username').text()).toEqual("");
    expect(wrapper.find('#password').text()).toEqual("");
    expect(wrapper.find('#passwordConfirmation').text()).toEqual("");
    expect(wrapper.find('#submitbtn1').exists()).toBeTruthy()
  });

  test('inputs should not be shown if not logged in', () => {
    expect(non_logged_in_wrapper.find('#name').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#lastName').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#email').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#emailConfirmation').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#username').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#password').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#passwordConfirmation').text()).toEqual("");
    expect(non_logged_in_wrapper.find('#submitbtn1').exists()).toBeTruthy()
  });

  test('validate if passwords do not match', () => {
    wrapper.find('#name').simulate('change', { target: { value: 'William', name: 'name' } })
    wrapper.find('#lastName').simulate('change', { target: { value: 'Ravelo', name: 'lastName' } })
    wrapper.find('#email').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'email' } })
    wrapper.find('#emailConfirmation').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'emailConfirmation' } })
    wrapper.find('#username').simulate('change', { target: { value: 'ravelinx', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '1234567', name: 'password' } })
    wrapper.find('#passwordConfirmation').simulate('change', { target: { value: '123456', name: 'passwordConfirmation' } })
    wrapper.find('#termsAndConditions').simulate('change', { target: { checked: true, name: 'termsAndConditions' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => {}
    });
    expect(wrapper.find('#error').text()).not.toEqual("");
  })


  test('validate if emails do not match', () => {
    wrapper.find('#name').simulate('change', { target: { value: 'William', name: 'name' } })
    wrapper.find('#lastName').simulate('change', { target: { value: 'Ravelo', name: 'lastName' } })
    wrapper.find('#email').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'email' } })
    wrapper.find('#emailConfirmation').simulate('change', { target: { value: 'drummerwilliam2@gmail.com', name: 'emailConfirmation' } })
    wrapper.find('#username').simulate('change', { target: { value: 'ravelinx', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#passwordConfirmation').simulate('change', { target: { value: '123456', name: 'passwordConfirmation' } })
    wrapper.find('#termsAndConditions').simulate('change', { target: { checked: true, name: 'termsAndConditions' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
    expect(wrapper.find('#error').text()).not.toEqual("");
  })

  test('validate if terms and conditions not accepted', () => {
    wrapper.find('#name').simulate('change', { target: { value: 'William', name: 'name' } })
    wrapper.find('#lastName').simulate('change', { target: { value: 'Ravelo', name: 'lastName' } })
    wrapper.find('#email').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'email' } })
    wrapper.find('#emailConfirmation').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'emailConfirmation' } })
    wrapper.find('#username').simulate('change', { target: { value: 'ravelinx', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#passwordConfirmation').simulate('change', { target: { value: '123456', name: 'passwordConfirmation' } })
    wrapper.find('#termsAndConditions').simulate('change', { target: { checked: false, name: 'termsAndConditions' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
    expect(wrapper.find('#error').text()).not.toEqual("");
  })

  test('register specialist', () => {
    axios.mockResolvedValueOnce(Promise.resolve({ data: { token: '' }}));

    wrapper.find('#name').simulate('change', { target: { value: 'William', name: 'name' } })
    wrapper.find('#lastName').simulate('change', { target: { value: 'Ravelo', name: 'lastName' } })
    wrapper.find('#email').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'email' } })
    wrapper.find('#emailConfirmation').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'emailConfirmation' } })
    wrapper.find('#username').simulate('change', { target: { value: 'ravelinx', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#passwordConfirmation').simulate('change', { target: { value: '123456', name: 'passwordConfirmation' } })
    wrapper.find('#termsAndConditions').simulate('change', { target: { checked: true, name: 'termsAndConditions' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
  })

  test('register specialist with error', () => {
    axios.mockResolvedValueOnce(Promise.reject({ response: { data: { mssg: '' } }}));

    wrapper.find('#name').simulate('change', { target: { value: 'William', name: 'name' } })
    wrapper.find('#lastName').simulate('change', { target: { value: 'Ravelo', name: 'lastName' } })
    wrapper.find('#email').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'email' } })
    wrapper.find('#emailConfirmation').simulate('change', { target: { value: 'drummerwilliam@gmail.com', name: 'emailConfirmation' } })
    wrapper.find('#username').simulate('change', { target: { value: 'ravelinx', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#passwordConfirmation').simulate('change', { target: { value: '123456', name: 'passwordConfirmation' } })
    wrapper.find('#termsAndConditions').simulate('change', { target: { checked: true, name: 'termsAndConditions' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
  })

  test('go to login', () => {
    wrapper.find('#goToLogin').simulate('click', {
      preventDefault: () => { }
    });
  })
})