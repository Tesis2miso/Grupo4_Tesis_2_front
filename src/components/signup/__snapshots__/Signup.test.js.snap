// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_Signup show signup 1`] = `
<Fragment>
  <form
    className="signup"
  >
    <h1
      className="title"
    >
      DermoApp
    </h1>
    <h2
      className="subtitle"
    >
      Registro de usuarios
    </h2>
    <br />
    <input
      className="formField"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      text=""
      type="text"
      value=""
    />
    <input
      className="formField"
      id="lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Last name"
      text=""
      type="text"
      value=""
    />
    <input
      className="formField"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      text=""
      type="email"
      value=""
    />
    <input
      className="formField"
      id="emailConfirmation"
      name="emailConfirmation"
      onChange={[Function]}
      placeholder="Email confirmation"
      text=""
      type="email"
      value=""
    />
    <input
      className="formField"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      text=""
      type="text"
      value=""
    />
    <input
      className="formField"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      text=""
      type="password"
      value=""
    />
    <input
      className="formField"
      id="passwordConfirmation"
      name="passwordConfirmation"
      onChange={[Function]}
      placeholder="Password confirmation"
      text=""
      type="password"
      value=""
    />
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="checkTermsAndConditions"
        id="termsAndConditions"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="conditionsLabel"
        htmlFor="termsAndConditions"
      >
        Do you accept terms and conditions?
      </label>
    </div>
    <label
      className="label_error"
      id="error"
    />
    <button
      className="goToLogin"
      id="goToLogin"
      onClick={[Function]}
    >
      ¿Ya tienes cuenta? Iniciar sesión
    </button>
    <button
      className="submitButton"
      id="submitbtn1"
      onClick={[Function]}
    >
      Registrarse
    </button>
  </form>
</Fragment>
`;
